Wikipedia/Code/UIImageView+WMFImageFetchingInternal.m

```diff
@@ -109,7 +109,8 @@ - (void)wmf_fetchImageDetectFaces:(BOOL)detectFaces onGPU:(BOOL)onGPU failure:(W
 
     UIImage *cachedImage = [self wmf_cachedImage];
     if (cachedImage) {
-        [self wmf_setImage:cachedImage detectFaces:detectFaces onGPU:onGPU animated:NO failure:failure success:success];
+        NSData *cachedImageData = cachedImage.isGIF ? [[self wmf_imageController] diskDataForImageWithURL:imageURL] : nil;
```

Comment:
This whole cached image check can be removed (lines 110-115) - the image load function below checks the cache as well.